legend-sdlc-extensions-collection-entity-serializer/src/test/java/org/finos/legend/sdlc/extensions/collection/entity/serializer/TestExtensions.java

```diff
@@ -0,0 +1,56 @@
+// Copyright 2021 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.sdlc.extensions.collection.entity.serializer;
+
+import org.finos.legend.sdlc.protocol.pure.v1.PureEntitySerializer;
+import org.finos.legend.sdlc.serialization.DefaultJsonEntitySerializer;
+import org.finos.legend.sdlc.serialization.EntitySerializer;
+import org.finos.legend.sdlc.serialization.EntitySerializers;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class TestExtensions
+{
+
+    @Test
+    public void testPlanGeneratorExtensionArePresent()
+    {
+        assertHasExtensions(EntitySerializers.getAvailableSerializers(), new HashSet<Class<? extends EntitySerializer>>(Arrays.asList(PureEntitySerializer.class, DefaultJsonEntitySerializer.class)), EntitySerializer.class, true);
+    }
+
+    private <T> void assertHasExtensions(Iterable<T> availableClasses, Set<? extends Class<? extends T>> expectedExtensionClasses, Class<T> extensionClass, boolean failOnAdditional)
+    {
+        List<Class<?>> unexpectedClasses = new ArrayList<>();
+        availableClasses.forEach(e ->
+        {
+            if (!expectedExtensionClasses.remove(e.getClass()))
+            {
+                unexpectedClasses.add(e.getClass());
+            }
+        });
+        Assert.assertEquals("Missing extensions for " + extensionClass.getName(), Collections.emptySet(), expectedExtensionClasses);
+        if (failOnAdditional)
+        {
+            Assert.assertEquals("Unexpected extensions for " + extensionClass.getName(), Collections.emptyList(), unexpectedClasses);
+        }
+    }
```

Comment:
Here's my suggestion:
```suggestion

    @Test
    public void testPlanGeneratorExtensionArePresent()
    {
        Map<String, EntitySerializer> serializersByName = EntitySerializers.getAvailableSerializersByName();
        
        Set<String> expectedNames = Stream.of("pure", "legend").collect(Collectors.toSet());
        Assert.assertEquals(expectedNames, serializersByName.keySet());
        
        Set<Class<? extends EntitySerializer>> expectedClasses = Stream.of(PureEntitySerializer.class, DefaultJsonEntitySerializer.class).collect(Collectors.ToSet());
        Set<Class<? extends EntitySerializer>> actualClasses = serializersByName.values().stream().map(EntitySerializer::getClass).collect(Collectors.toSet());
        Assert.assertEquals(expectedClasses, actualClasses);
    }
```